app/jobs/autoscaling.py

```diff
@@ -74,83 +74,90 @@ def autoscaling_update(config):
+                    # Note: safety here relies on the fact that the pipeline monitor is single-threaded, so no new stages can be dispatched
+                    # between the 'num_chunks == 0' measurement and the discardment of the instances. The web app is also unable to dispatch any
+                    # stage 2 jobs by itself -- it can only dispatch stage 1 jobs (new sample uploads), which do not call gsnap/rapsearch ASGs until
+                    # the pipeline_monitor moves them along to stage 2.
+                    service_ASG.remove_scalein_protection(available_instances + draining_instances)
+                    continue
+
+            print "AVAILABLE <--> DRAINING transitions:"
+            service_ASG.draining_instances = draining_instances
+            if new_num_desired < num_available:
+                num_to_drain = num_available - new_num_desired
+                print "{num_to_drain} instances need to be drained.".format(num_to_drain=num_to_drain)
+                instances_to_drain = service_ASG.start_draining(num_to_drain)
+                print "Moved {instances_to_drain} from 'available' to 'draining' state.".format(instances_to_drain=instances_to_drain)
+            elif new_num_desired > num_available and num_draining > 0:
+                num_to_rescue = min(new_num_desired - num_available, num_draining)
+                print "{num_to_rescue} instances need to stop draining.".format(num_to_rescue=num_to_rescue)
+                instances_to_rescue = service_ASG.stop_draining(num_to_rescue)
+                print "Moved {instances_to_rescue} from 'draining' to 'available' state.".format(instances_to_rescue=instances_to_rescue)
+            else:
+                print "No instances need to make an AVAILABLE <--> DRAINING transition."
+
+            print "DRAINING --> DISCARDED transitions:"
+            instances_to_discard = service_ASG.discard_if_safe()
+            num_to_discard = len(instances_to_discard)
+            if num_to_discard > 0:
+                print "{num_to_discard} instances have finished draining and can be discarded: {instances_to_discard}.".format(num_to_discard=num_to_discard, instances_to_discard=instances_to_discard)
+            else:
+                print "No instances are ready to move from 'draining' to 'discarded' state."
+
+        if 'batch_configurations' in config:
+            print "Autoscaling batches:"
+            response_batch_autoscaling = batch_autoscaling.autoscale_compute_environments(config['batch_configurations'])
+            print json.dumps(response_batch_autoscaling, default=_default_json_serializer)
+    except Exception as error:
+        # If any error occurs while performing an autoscaling update, append [Datadog] to ensure that the log is
+        # forwarded to Datadog.
+        # TODO(julie): Note that this is part of a temporary stopgap measure as we move away from Datadog and should be
```

Comment:
could you date this comment and also include the JIRA number?

Reply:
could you also `raise error` again at the end of this? in case someone ever adds more handling around this

Reply:
Done!